cocos/renderer/renderer/Pass.h

```diff
@@ -175,7 +176,7 @@ class Pass : public Ref
     Pass* _parent = nullptr;
     
     std::unordered_map<size_t, Technique::Parameter> _properties;
-    ValueMap _defines;
```

Comment:
The iteration order of `std::unordered_map` is not stable. `Pass::generateDefinesKey()` can generate different hash for the same `defines` settings.